db/warehouse/migrate/20231020151224_add_client_merge_histories_table.rb

```diff
@@ -0,0 +1,11 @@
+class AddClientMergeHistoriesTable < ActiveRecord::Migration[6.1]
+  def change
+    create_table :hmis_client_merge_histories do |t|
+      t.references :retained_client, null: true, foreign_key: { to_table: 'Client' }, index: true
```

Comment:
Do you want some cascade delete in here?  I worry about these foreign key constraints, probably less of a problem in the HMIS, but in the warehouse data is often deleted in batches without regard for dependent data.

Reply:
Wouldnt the cascade delete go on the Hmis::Hud::Client model class?
I guess I'm not sure. I don't want things to blow up if a client is deleted. When I put `null: true` I think that's what I was thinking, but it doesn't make sense - the value itself should never be null, but its possible that the record doesnt exist, because of what you're saying.

So should it just be like this:
```
t.references :retained_client, null: false, index: true
t.references : deleted_client, null: false
```

Reply:
that will work (wit a fix for the space on the second line).  I think you should be able to do something like

`add_foreign_key :my_object_times, :my_objects, on_delete: :cascade`
maybe
`t.references :retained_client, null: true, foreign_key: { to_table: 'Client' }, index: true, on_delete: :cascade`
but I haven't tested it.